base/src/AffineTransform.cpp

```diff
@@ -0,0 +1,398 @@
+			rawOutMetadata->setData(outputMetadata);
+			imageType = rawMetadata->getImageType();
+			depth = rawMetadata->getDepth();
+		}
+
+		switch (imageType)
+		{
+		case ImageMetadata::MONO:
+			if (depth != CV_8U)
+			{
+				throw AIPException(AIP_NOTIMPLEMENTED, "Rotate not supported for bit depth<" + std::to_string(depth) + ">");
+			}
+			break;
+		case ImageMetadata::BGR:
+		case ImageMetadata::RGB:
+		case ImageMetadata::RGBA:
+		case ImageMetadata::BGRA:
+		case ImageMetadata::YUV444:
+		case ImageMetadata::YUV420:
+			if (depth != CV_8U)
+			{
+				throw AIPException(AIP_NOTIMPLEMENTED, "Rotate not supported for bit depth<" + std::to_string(depth) + ">");
+			}
+			break;
+		}
+
+		mFrameLength = mOutputMetadata->getDataSize();
+		setMetadataHelper(metadata, mOutputMetadata);
+		double si, co;
+		si = sin(props.angle * PI / 180);
+		co = props.scale * cos(props.angle * PI / 180);
+		acoeff[0][0] = co;
+		acoeff[0][1] = -si;
+		acoeff[0][2] = props.x;
+		acoeff[1][0] = si;
+		acoeff[1][1] = co;
+		acoeff[1][2] = props.y;
+	}
+
+	bool compute(void *buffer, void *outBuffer)
+	{
+		auto status = NPP_SUCCESS;
+		auto bufferNPP = static_cast<Npp8u *>(buffer);
+		auto outBufferNPP = static_cast<Npp8u *>(outBuffer);
+		if (mFrameType == FrameMetadata::RAW_IMAGE_PLANAR)
+		{
+			for (auto i = 0; i < channels; i++)
+			{
+				src[i] = bufferNPP + srcNextPtrOffset[i];
+				dst[i] = outBufferNPP + dstNextPtrOffset[i];
+
+				status = nppiWarpAffine_8u_C1R_Ctx(src[i],
+												   srcSize[i],
+												   srcPitch[i],
+												   srcRect[i],
+												   dst[i],
+												   dstPitch[i],
+												   dstRect[i],
+												   acoeff,
+												   NPPI_INTER_NN,
+												   nppStreamCtx);
+			}
+		}
+
+		if (mFrameType == FrameMetadata::RAW_IMAGE)
+		{
+
+			if (channels == 1 && depth == CV_8UC1)
+			{
+				status = nppiWarpAffine_8u_C1R_Ctx(const_cast<const Npp8u *>(bufferNPP),
+												   srcSize[0],
+												   srcPitch[0],
+												   srcRect[0],
+												   outBufferNPP,
+												   dstPitch[0],
+												   dstRect[0],
+												   acoeff,
+												   NPPI_INTER_NN,
+												   nppStreamCtx);
+			}
+			else if (channels == 3)
+			{
+				status = nppiWarpAffine_8u_C3R_Ctx(const_cast<const Npp8u *>(bufferNPP),
+												   srcSize[0],
+												   srcPitch[0],
+												   srcRect[0],
+												   outBufferNPP,
+												   dstPitch[0],
+												   dstRect[0],
+												   acoeff,
+												   NPPI_INTER_NN,
+												   nppStreamCtx);
+			}
+			else if (channels == 4)
+			{
+				status = nppiWarpAffine_8u_C4R_Ctx(const_cast<const Npp8u *>(bufferNPP),
+												   srcSize[0],
+												   srcPitch[0],
+												   srcRect[0],
+												   outBufferNPP,
+												   dstPitch[0],
+												   dstRect[0],
+												   acoeff,
+												   NPPI_INTER_NN,
```

Comment:
The interpolation method used should be a module property with the defaul being "NPPI_INTER_NN"

Reply:
resolved this issue made interpolation method as an module property